lib/light-service/organizer.rb

```diff
@@ -56,6 +56,14 @@ def execute(code_block)
       def with_callback(action, steps)
         WithCallback.run(self, action, steps)
       end
+
+      def log_with(logger)
+        @logger = logger
+      end
+
+      def logger
+        @logger
```

Comment:
What do you think about moving the conditional into here?

```ruby
def logger
  @logger || LightService::Configuration.logger
end
```
That way  we can use `logger = monitored_organizer.logger` without further conditionals in `lib/light-service/organizer/with_reducer_factory.rb`.

Reply:
Yes, there will be less conditionals in `WithReducerFactory` but the object will be a little bit less "functional" -- if the global logger is changed, the way the classes behave will change during runtime. One could argue that this is desirable.

For the way I use it, it's not a problem, so I defer the decision to you: which do you prefer?

Reply:
I like your suggestion, keep it as is. Thanks!